utils/pinprocfw/pinprocfw.cpp

```diff
@@ -2176,10 +2175,20 @@ int checkPROCFile() {
     PRReadData(proc, 0, 0, 4, readdata);
     board_id = readdata[0];
     board_rev = readdata[3];
-    board_rev = (board_rev & 0x80) >> 7 |
-                (board_rev & 0x40) >> 5 |
-                (board_rev & 0x20) >> 3 |
-                (board_rev & 0x10) >> 1;
+    if (board_id == P3_ROC_CHIP_ID) {
+        board_rev = (board_rev & 0x800) >> 11 |
+                    (board_rev & 0x400) >> 10 |
+                    (board_rev & 0x200) >> 9 |
+                    (board_rev & 0x100) >> 8;
+        fprintf(stderr, "\nReading P3-ROC board_rev: %d", board_rev);
+    }
+    else {
+        board_rev = (board_rev & 0x80) >> 7 |
+                    (board_rev & 0x40) >> 6 |
+                    (board_rev & 0x20) >> 5 |
+                    (board_rev & 0x10) >> 4;
+        fprintf(stderr, "\nReading P-ROC board_rev: %d", board_rev);
+    }
```

Comment:
This is @gstellenberg's addition for correctly calculating the board revision for P3-ROC and P-ROC.